src/TraceEvent/EventPipe/EventPipeMetadata.cs

```diff
@@ -462,112 +463,112 @@ private DynamicTraceEventData.PayloadFetch ParseType(
             {
                 case EventPipeTypeCode.Boolean32:
                     {
-                        payloadFetch.Type = typeof(bool);
+                        payloadFetch.Type = FetchType.System_Boolean;
```

Comment:
Can you please explain why this change is necessary?  I'm not arguing that it isn't necessary - I just don't understand enough about NativeAOT to know and would like to educate myself.